components/prism-php.js

```diff
@@ -42,14 +49,19 @@
 		}
 	});
 
-	Prism.languages.insertBefore('php', 'keyword', {
+	// Must be defined before class-name (because primitive types must be matched before class names in function typehint)
+	Prism.languages.insertBefore('php', 'class-name', {
 		'variable': /\$+(?:\w+\b|(?={))/i,
 		'package': {
-			pattern: /(\\|namespace\s+|use\s+)[\w\\]+/,
+			pattern: /(\\|namespace\s+|use\s+(function\s+)*)[a-zA-Z_][\w\\]+/,
 			lookbehind: true,
 			inside: {
 				punctuation: /\\/
 			}
+		},
+		'type': {
+			pattern: /(bool|boolean|int|integer|float|string|object|void|array|mixed)(?!_)/,
```

Comment:
1. Please use non-capturing groups (one of the reasons the tests fail).
1. Please use boundary assertions.

```suggestion
			pattern: /\b(?:bool|boolean|int|integer|float|string|object|void|array|mixed)\b/,
```

Reply:
fixed

Reply:
btw `boolean` and `integer` are not valid typehints in PHP (they are understood as class names), only `bool` and `int`. In PHP 8.0 is valid typehint also `false` (but not `true`).

Reply:
You are right, i will fix it in PHP8 features PR https://github.com/PrismJS/prism/pull/2591